dali/operators/geometry/affine_transforms/transform_scale.cc

```diff
@@ -80,7 +90,20 @@ class TransformScaleCPU
   void ProcessArgs(const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
     assert(scale_.IsDefined());
     scale_.Acquire(spec, ws, nsamples_, true);
-    ndim_ = scale_[0].num_elements();
+    int scale_ndim = scale_[0].num_elements();
+
+    if (scale_ndim > 1) {
+      ndim_ = scale_ndim;
+    } else if (has_input_) {
+      ndim_ = input_transform_ndim(ws);
+    } else if (ndim_arg_ > 0) {
+      ndim_ = ndim_arg_;
+    } else {
+      ndim_ = 1;
+    }
+    DALI_ENFORCE(scale_ndim == ndim_ || scale_ndim == 1,
```

Comment:
If scale_ndim > 1: scale_ndim == ndim_ is always true because of L96.
If scale_ndim == 1 this case applies
If  scale_ndim < 1 this will fail as ndim_  cannot be 0, right?
So the condition should be rather:
```
DALI_ENFORCE(scale_ndim > 0,
```

Reply:
Not really. What I am checking here is:
- I have either a scalar and it is meant to be a scalar -> nothing to do
- I have a scalar and it is meant to be ndim -> propagate
- I have scale_ndim > 1 and ndim > 1 and they both match

scale_ndim_ != ndim_ would be an error, except for the case when scale_ndim is a scalar

Reply:
Not really. What I am checking here is:
- I have either a scalar and it is meant to be a scalar -> nothing to do
- I have a scalar and it is meant to be ndim -> propagate
- I have scale_ndim > 1 and ndim > 1 and they both match

scale_ndim_ != ndim_ would be an error, except for the case when scale_ndim is a scalar

Reply:
scale_ndim will never be < 0, so I guess the check is not really necessary.

Reply:
removed